Modules/_xxsubinterpretersmodule.c

```diff
@@ -2366,6 +2531,50 @@ PyDoc_STRVAR(channel_recv_doc,
 \n\
 Return a new object from the data at the from of the channel's queue.");
 
+static PyObject *
+channel_send_wait(PyObject *self, PyObject *args, PyObject *kwds)
+{
+    static char *kwlist[] = {"cid", "obj", "timeout", NULL};
+    int64_t cid;
+    PyObject *obj;
+    int64_t timeout = -1;
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O&O|l:channel_send_wait",
+                                     kwlist, channel_id_converter,
+                                     &cid, &obj, &timeout)) {
+        return NULL;
+    }
+
+    void *wait_lock = _channelitem_wait_lock_new();
+    if (_channel_send(&_globals.channels, cid, obj, wait_lock) != 0) {
+        return NULL;
+    }
+
+    long long microseconds;
+    if (timeout >= 0) {
+        microseconds = timeout * 1000000;
+    }
+    else {
+        microseconds = -1;
+    }
+    PyLockStatus lock_rc = _channelitem_wait_lock_wait(wait_lock, microseconds);
+
+    if (lock_rc == PY_LOCK_ACQUIRED) {
+        Py_RETURN_TRUE;
+    }
+    else {
+        Py_RETURN_FALSE;
+    }
```

Comment:
We've tried to avoid putting lower-level code like this in the actual extension module functions.  Either move it up into `_channel_send()` (my preference) or into a new `_channel_send_wait()`.  Then this code would look more like the following:

```suggestion
    long long microseconds = timeout * 1000000;
    if (_channel_send(&_globals.channels, cid, obj, timeout) != 0) {
        if (PyErr_Occurred()) {
            return NULL;
        }
        Py_RETURN_FALSE;
    }
    Py_RETURN_TRUE;
```